b2share/modules/records/api.py

```diff
@@ -0,0 +1,128 @@
+            'checksum': self.obj.file.checksum,
+            'key': self.obj.key,  # IMPORTANT it must stay here!
+            'size': self.obj.file.size,
+            'version_id': str(self.obj.version_id),
+            'b2safe_pid': self.b2safe_pid,
+        })
+        return self.data
+
+
+class B2ShareRecord(Deposit, Record):
+    """B2Share record class."""
+    file_cls = B2ShareFileObject
+
+    @classmethod
+    def delete(cls, pid, record):
+        """Delete a published record.
+
+        params:
+        pid: PersistentIdentifier
+        record: invenio_records_files.api.Record instance
+        """
+        # Fetch deposit id from record and resolve deposit record and pid.
+        depid = b2share_deposit_uuid_fetcher(None, record)
+        if not depid:
+            abort(404)
+        depid, deposit = Resolver(
+            pid_type=depid.pid_type,
+            object_type='rec',
+            # Retrieve the deposit with the Record class on purpose as the current
+            # Deposit api prevents the deletion of published deposits.
+            getter=Record.get_record,
+        ).resolve(depid.pid_value)
+
+        # Note: the record is unindexed from Elasticsearch via the
+        # signal handlers in triggers.py
+        try:
+            RecordIndexer().delete(deposit)
+        except NotFoundError:
+            pass
+
+        # remove record from versioning chain
+        # if the parent has no other children and no draft child
+        # delete it from the db completely
+        version_child = PIDVersioning(child=pid)
+        version_master = PIDVersioning(parent=version_child.parent)
+        if version_master is not None:
+            version_master.remove_child(pid)
+            if not version_master.children.all() and \
+                    not version_master.draft_child:
+                db.session.delete(version_master.parent)
+            else:
+                new_last_child_rec_meta = RecordMetadata.query.filter_by(
+                    id=version_master.last_child.object_uuid).one_or_none()
+                new_last_child_rec = Record(new_last_child_rec_meta.json,
+                                            model=new_last_child_rec_meta)
+                RecordIndexer().index(new_last_child_rec)
+
+        # Mark connected buckets as DELETED
+        if record.files:
+            record_bucket = record.files.bucket
+            deposit_bucket = deposit.files.bucket
+            Bucket.delete(deposit_bucket.id)
+            Bucket.delete(record_bucket.id)
+
+        # mark all PIDs as DELETED
+        all_pids = PersistentIdentifier.query.filter(
+            PersistentIdentifier.object_type == pid.object_type,
+            PersistentIdentifier.object_uuid == pid.object_uuid,
+        ).all()
+        for rec_pid in all_pids:
+            if not rec_pid.is_deleted():
+                rec_pid.delete()
+        depid.delete()
+
+        # Mark the record and deposit as deleted
+        record.delete()
```

Comment:
config's `B2SHARE_RECORDS_REST_ENDPOINTS.record_class` should be changed to use this record class. Then we should call `super` here.